src/compiler/checker.ts

```diff
@@ -11275,42 +11305,41 @@ namespace ts {
          * property is computed by inferring from the source property type to X for the type
          * variable T[P] (i.e. we treat the type T[P] as the type variable we're inferring for).
          */
-        function inferTypeForHomomorphicMappedType(source: Type, target: MappedType, mappedTypeStack: string[]): Type {
+        function inferTypeForHomomorphicMappedType(source: Type, target: MappedType): Type {
+            const key = source.id + "," + target.id;
+            if (deferredInferenceCache.has(key)) {

```

Comment:
Do we actually need this cache?

Reply:
Yep, turns out some of the tests run out of memory without it.